onnx/defs/tensor/defs.cc

```diff
@@ -170,6 +171,23 @@ Example 2:
             "Constrain input and output types to float tensors.");
 
 ONNX_OPERATOR_SCHEMA(Transpose)
+    .TensorInferenceFunction([](Node* n) {
+      ONNX_ASSERT(n->kind() == kTranspose)
```

Comment:
I think we can assume that dispatch mechanism does the right thing. There might be some ops without associated constant type in ir.h and we should be able to still handle those